lib/blocks.php

```diff
@@ -374,8 +374,10 @@ function gutenberg_register_theme_block_category( $categories ) {
 	);
 	return $categories;
 }
-
-add_filter( 'block_categories', 'gutenberg_register_theme_block_category' );
+// This can be removed when plugin support requires WordPress 5.8.0+.
+if ( ! function_exists( 'get_default_block_categories' ) ) {
+	add_filter( 'block_categories', 'gutenberg_register_theme_block_category' );
```

Comment:
The question I have is whether these should be deprecated in the first place. I know you've thought about the `_all`  prefix to extend all screens at the same time, did you consider using these existing filters for that use case?

Reply:
Do you mean to use `block_categories`? It has the `$post` object as a param which is not set outside of the edit post page. We even provide examples for this filter that depend on `$post` at https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/filters/block-filters.md#managing-block-categories:

```php
<?php
// my-plugin.php

function my_plugin_block_categories( $categories, $post ) {
	if ( $post->post_type !== 'post' ) {
		return $categories;
	}
	return array_merge(
		$categories,
		array(
			array(
				'slug' => 'my-category',
				'title' => __( 'My category', 'my-plugin' ),
				'icon'  => 'wordpress',
			),
		)
	);
}
add_filter( 'block_categories', 'my_plugin_block_categories', 10, 2 );
```

If you have some ideas on how to address it safely then we should do it.

Reply:
mmm right :) if it's already receiving `$post`, it's not a good fit for the "all" use-case.

Reply:
@jeremyfelt raised the same issue in his comment on WordPress trac: https://core.trac.wordpress.org/ticket/52920#comment:24.